File: app/src/main/java/com/example/iteventscheckin/network/EventApi.java
package com.example.iteventscheckin.network;

import com.example.iteventscheckin.models.Event;
import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Query;

import java.util.List;

public interface EventApi {
    @GET("/api/v1/Events/registration")
    Single<List<Event>> getAllEvents(
            @Query("token") String token);
}

File: app/src/main/java/com/example/iteventscheckin/network/EventApi.kt
package com.example.iteventscheckin.network

import com.example.iteventscheckin.models.Event
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Query

interface EventApi {
    @GET("/api/v1/Events/registration")
    fun getAllEvents(
        @Query("token") token: String
    ): Single<List<Event>>
}
